Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -2287,6 +2360,93 @@ public AndConstraint<TAssertions> NotContainInOrder(params T[] unexpected)
+    /// </summary>
+    /// <remarks>
+    /// Elements are compared using their <see cref="object.Equals(object)" /> implementation.
+    /// </remarks>
+    /// <param name="unexpected">A <see cref="Array"/> with the unexpected elements.</param>
+    /// <exception cref="ArgumentNullException"><paramref name="unexpected"/> is <c>null</c>.</exception>
+    public AndConstraint<TAssertions> NotContainInConsecutiveOrder(params T[] unexpected)
+    {
+        return NotContainInConsecutiveOrder(unexpected, string.Empty);
+    }
+
+    /// <summary>
+    /// Asserts the current collection does not contain the specified elements in the exact same order and consecutively.
+    /// </summary>
+    /// <remarks>
+    /// Elements are compared using their <see cref="object.Equals(object)" /> implementation.
+    /// </remarks>
+    /// <param name="unexpected">An <see cref="IEnumerable{T}"/> with the unexpected elements.</param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    /// <exception cref="ArgumentNullException"><paramref name="unexpected"/> is <c>null</c>.</exception>
+    public AndConstraint<TAssertions> NotContainInConsecutiveOrder(IEnumerable<T> unexpected, string because = "",
+        params object[] becauseArgs)
+    {
+        Guard.ThrowIfArgumentIsNull(unexpected, nameof(unexpected), "Cannot verify absence of ordered containment against a <null> collection.");
+
+        if (Subject is null)
+        {
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Cannot verify absence of ordered containment in a <null> collection.");
+
+            return new AndConstraint<TAssertions>((TAssertions)this);
+        }
+
+        IList<T> unexpectedItems = unexpected.ConvertOrCastToList();
+        if (unexpectedItems.Any())
+        {
+            IList<T> actualItems = Subject.ConvertOrCastToList();
+
+            if (unexpectedItems.Count > actualItems.Count)
+            {
+                return new AndConstraint<TAssertions>((TAssertions)this);
+            }
+
+            int subjectIndex = 0;
+
+            Func<T, T, bool> areSameOrEqual = ObjectExtensions.GetComparer<T>();
+            int index;
+
+            for (index = 0; index < unexpectedItems.Count; index++)
+            {
+                T unexpectedItem = unexpectedItems[index];
+
+                int previousSubjectIndex = subjectIndex;
+                subjectIndex = IndexOf(actualItems, unexpectedItem, subjectIndex, areSameOrEqual);
```

Comment:
♻️ I think we can make this call more self-explanatory by doing two things:
1. Since all of the invocations pass the same compared, make this method generic and then have it call `ObjectExtensions.GetComparer`.
2. Rename the `index` parameter to `startIndex` and then use a named parameter on the calling side. I had to study the implementation to understand what it did.